go.mod

```diff
@@ -1,57 +1,53 @@
 module github.com/hashicorp/hcp-scada-provider
 
-go 1.18
+go 1.23.0
```

Comment:
I think some of our tests are failing due to 1.18 still being specified here: https://github.com/hashicorp/hcp-scada-provider/blob/main/.github/workflows/test.yml#L17